pkg/collector/corechecks/system/disk_nix.go

```diff
@@ -0,0 +1,154 @@
+		usage, err := diskUsage(partition.Mountpoint)
+		if err != nil {
+			log.Warnf("Unable to get disk metrics of %s mount point: %s", partition.Mountpoint, err)
+			continue
+		}
+
+		// Exclude disks with total disk size 0
+		if usage.Total == 0 {
+			continue
+		}
+
+		tags := make([]string, len(c.cfg.customeTags), len(c.cfg.customeTags)+2)
+		copy(tags, c.cfg.customeTags)
+
+		if c.cfg.tagByFilesystem {
+			tags = append(tags, fmt.Sprintf("filesystem:%s", partition.Fstype))
+		}
+		var deviceName string
+		if c.cfg.useMount {
+			deviceName = partition.Mountpoint
+		} else {
+			deviceName = partition.Device
+		}
+		tags = append(tags, fmt.Sprintf("device:%s", deviceName))
+
+		tags = c.applyDeviceTags(partition.Device, partition.Mountpoint, tags)
+
+		c.sendPartitionMetrics(sender, usage, tags)
+	}
+
+	return nil
+}
+
+func (c *DiskCheck) collectDiskMetrics(sender aggregator.Sender) error {
+	iomap, err := ioCounters()
+	if err != nil {
+		return err
+	}
+	for deviceName, ioCounter := range iomap {
+
+		tags := make([]string, len(c.cfg.customeTags), len(c.cfg.customeTags)+1)
+		copy(tags, c.cfg.customeTags)
+		tags = append(tags, fmt.Sprintf("device:%s", deviceName))
+
+		tags = c.applyDeviceTags(deviceName, "", tags)
+
+		c.sendDiskMetrics(sender, ioCounter, tags)
+	}
+
+	return nil
+}
+
+func (c *DiskCheck) sendPartitionMetrics(sender aggregator.Sender, usage *disk.UsageStat, tags []string) {
+	// Disk metrics
+	// For legacy reasons,  the standard unit it kB
+	sender.Gauge(fmt.Sprintf(diskMetric, "total"), float64(usage.Total)/1024, "", tags)
+	sender.Gauge(fmt.Sprintf(diskMetric, "used"), float64(usage.Used)/1024, "", tags)
+	sender.Gauge(fmt.Sprintf(diskMetric, "free"), float64(usage.Free)/1024, "", tags)
+	sender.Gauge(fmt.Sprintf(diskMetric, "in_use"), usage.UsedPercent/100, "", tags)
+	// Use percent, a lot more logical than in_use
+	sender.Gauge(fmt.Sprintf(diskMetric, "used.percent"), usage.UsedPercent, "", tags)
+
+	// Inodes metrics
+	sender.Gauge(fmt.Sprintf(inodeMetric, "total"), float64(usage.InodesTotal), "", tags)
+	sender.Gauge(fmt.Sprintf(inodeMetric, "used"), float64(usage.InodesUsed), "", tags)
+	sender.Gauge(fmt.Sprintf(inodeMetric, "free"), float64(usage.InodesFree), "", tags)
+	sender.Gauge(fmt.Sprintf(inodeMetric, "in_use"), usage.InodesUsedPercent/100, "", tags)
+	// Use percent, a lot more logical than in_use
+	sender.Gauge(fmt.Sprintf(inodeMetric, "used.percent"), usage.InodesUsedPercent, "", tags)
+
+}
+
+func (c *DiskCheck) sendDiskMetrics(sender aggregator.Sender, ioCounter disk.IOCountersStat, tags []string) {
+
+	// /1000 as psutil returns the value in ms
+	// Rate computes a rate of change between to consecutive check run.
+	// For cumulated time values like read and write times this a ratio between 0 and 1, we want it as a percentage so we *100 in advance
+	sender.Rate(fmt.Sprintf(diskMetric, "read_time_pct"), float64(ioCounter.ReadTime)*100/1000, "", tags)
```

Comment:
The python version does not report this metrics for `sda` but does for it's partition, `sda1`. Yours does for both. That's probably fine 🤔

Reply:
It seems to be a difference between psutil and gopsutil, I think it's fine since both libraries actually read the content of `/proc/diskstats`